### PR TITLE
Homework20. JDBC.

### PR DESCRIPTION
**Pull request by Cherkashyn**

### CODE REVIEW

File: homework1/src/main/java/lesson20/OfficeDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public Offices getOfficeById(BigDecimal officeId) throws SQLException {
+        Offices offices = null;
+
+        Connection connection = ConnectionDB.getConnection();
+        PreparedStatement statement = connection.prepareStatement("select * from offices where office = ?");
+        statement.setBigDecimal(1, officeId);
+        ResultSet resultSet = statement.executeQuery();
+
+        while (resultSet.next()){
+            offices = new Offices(officeId, resultSet.getString("CITY"), resultSet.getString("REGION"), resultSet.getBigDecimal("MGR"), resultSet.getBigDecimal("TARGET"), resultSet.getBigDecimal("SALES"));
```
Comment (Yanetta) [inline homework1/src/main/java/lesson20/OfficeDaoImpl.java]: I think, officeId should be returned from resultSet.

File: homework1/src/main/java/lesson20/OfficeDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        resultSet.close();
+        statement.close();
+        connection.close();
+
+        return officesSet;
+    }
+
+    @Override
+    public Offices getOfficeById(BigDecimal officeId) throws SQLException {
```
Comment (Yanetta) [inline homework1/src/main/java/lesson20/OfficeDaoImpl.java]: As I understood from HW assignment , this method should return several objects.

File: homework1/src/main/java/lesson20/OrderDaoImpl.java [hunk 3]
```diff
@@ -28,10 +26,65 @@ public Order findOrderById(BigDecimal id) throws SQLException{
...
+        Product product = null;
+
+        Connection connection = ConnectionDB.getConnection();
+        Statement orderStatement = connection.createStatement();
+        ResultSet orderRs = orderStatement.executeQuery("select * from orders");
+
+        while (orderRs.next()){
+            order = new Order(orderRs.getBigDecimal("ORDER_NUM"), orderRs.getDate("ORDER_DATE"), orderRs.getString("MFR"), product, orderRs.getBigDecimal("QTY"), orderRs.getBigDecimal("AMOUNT"));
+
+            PreparedStatement productStatement = connection.prepareStatement("select * from products where product_id = ?");
```
Comment (Yanetta) [inline homework1/src/main/java/lesson20/OrderDaoImpl.java]: There is a possibility, that in one connection we should make one statement, but I'm not sure. I also made two statements)))

File: homework1/src/test/java/lesson20/OrderDaoImplIntegrationTest.java [hunk 4]
```diff
@@ -8,11 +8,16 @@
 
 import java.math.BigDecimal;
 import java.sql.SQLException;
+import java.util.Set;
 
 @FixMethodOrder(MethodSorters.NAME_ASCENDING)
 public class OrderDaoImplIntegrationTest {
```
Comment (Yanetta) [inline homework1/src/test/java/lesson20/OrderDaoImplIntegrationTest.java]: May be tests for offices should be included in a separate test class.

Comment (Cherkashyn) [reply homework1/src/test/java/lesson20/OrderDaoImplIntegrationTest.java]: yeap. I'll change it.

File: homework1/src/test/java/lesson20/OrderDaoImplIntegrationTest.java [hunk 5]
```diff
@@ -8,11 +8,16 @@
 
 import java.math.BigDecimal;
 import java.sql.SQLException;
+import java.util.Set;
 
 @FixMethodOrder(MethodSorters.NAME_ASCENDING)
```
Comment (SergeyPasko) [inline homework1/src/test/java/lesson20/OrderDaoImplIntegrationTest.java]: Why you use that?

File: homework1/src/main/java/lesson20/OfficeDaoImpl.java (hunk 2 above)
Comment (SergeyPasko) [reply homework1/src/main/java/lesson20/OfficeDaoImpl.java]: Please change it

File: homework1/src/test/java/lesson20/OrderDaoImplIntegrationTest.java [hunk 6]
```diff
@@ -21,4 +26,32 @@ public void testFindOrderById() throws SQLException{
         assertNotNull(order);
     }
 
+    @Test
+    public void testGetAllOrders() throws SQLException{
```
Comment (SergeyPasko) [inline homework1/src/test/java/lesson20/OrderDaoImplIntegrationTest.java]: Also good test - cheak that both metods return same result

File: homework1/src/test/java/lesson20/OfficesDaoImplIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+public class OfficesDaoImplIntegrationTest {
+    private static final BigDecimal ALREADY_EXIST_OFFICE = BigDecimal.valueOf(22);
+    private OfficesDao officesDao = new OfficeDaoImpl();
+
+    @Test
+    public void testGetAllOffices() throws SQLException {
+        Set<Offices> officesSet = officesDao.getAllOffices();
+        System.out.println(officesSet);
+        assertNotNull(officesSet);
```
Comment (SergeyPasko) [inline homework1/src/test/java/lesson20/OfficesDaoImplIntegrationTest.java]: This don't cheack anything. You always have notNullSet in result

File: homework1/src/test/java/lesson20/OrderDaoImplIntegrationTest.java [hunk 8]
```diff
@@ -21,4 +23,18 @@ public void testFindOrderById() throws SQLException{
...
+        Set<Order> orderSet = orderDao.getAllOrders();
+        System.out.println(orderSet);
+        assertNotNull(orderSet);
+    }
+
+    @Test
+    public void testGetAllOrdersByJoin() throws SQLException{
+        Set<Order> orderSet = OrderDaoImpl.getAllOrdersByJoin();
+        System.out.println(orderSet);
+        assertNotNull(orderSet);
```
Comment (SergeyPasko) [inline homework1/src/test/java/lesson20/OrderDaoImplIntegrationTest.java]: Also

File: homework1/src/main/java/lesson20/OfficeDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        resultSet.close();
+        statement.close();
+        connection.close();
+
+        return officesSet;
+    }
+
+    @Override
+    public Set<Offices> getOfficeById(BigDecimal officeId) throws SQLException {
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson20/OfficeDaoImpl.java]: What logic? In this case you can have only 0..1 results in Set

**Pull request merged**